subsys/bluetooth/controller/ll_sw/ull_adv_sync.c

```diff
@@ -172,6 +179,16 @@ static uint8_t adv_sync_pdu_init_from_prev_pdu(struct pdu_adv *pdu,
 		dptr_prev += sizeof(struct pdu_cte_info);
 	}
 
+	if (IS_ENABLED(CONFIG_BT_CTLR_ADV_PERIODIC_ADI_SUPPORT) && ext_hdr->adi != 0) {
+		if (ext_hdr_prev->adi) {
+			memcpy(dptr, dptr_prev, sizeof(struct pdu_adv_adi));
+		}
+		dptr += sizeof(struct pdu_adv_adi);
+	}
+	if (ext_hdr_prev->adi) {
+		dptr_prev += sizeof(struct pdu_adv_adi);
+	}
```

Comment:
I'm not sure I follow this completely: 
So if the `ext_hdr_prev` has `adi`, then we copy it to the `dptr` from `dptr_prev`, and increment `dptr`, but then we again check if `ext_hdr_prev` has `adi` and increment the `dptr_prev`. 

Should the first `if` have been `ext_hdr`, instead of `ext_hdr_prev`?

Reply:
`ext_hrd->adi != 0` is in line 182. It that what you are asking for?

Reply:
Ah, yes, missed that :)